x/oracle/keeper/keeper_test.go

```diff
@@ -0,0 +1,255 @@
+	s.Require().NoError(app.BankKeeper.SendCoinsFromModuleToAccount(ctx, minttypes.ModuleName, addr, initCoins))
+	_, err := sh(ctx, NewTestMsgCreateValidator(valAddr, valPubKey, amt))
+	s.Require().NoError(err)
+
+	staking.EndBlocker(ctx, app.StakingKeeper)
+
+	s.app = app
+	s.ctx = ctx
+	s.queryClient = types.NewQueryClient(queryHelper)
+}
+
+// Test addresses
+var (
+	valPubKeys = simapp.CreateTestPubKeys(1)
+	valPubKey  = valPubKeys[0]
+	pubKey     = secp256k1.GenPrivKey().PubKey()
+	addr       = sdk.AccAddress(pubKey.Address())
+	valAddr    = sdk.ValAddress(pubKey.Address())
+
+	initTokens = sdk.TokensFromConsensusPower(initialPower, sdk.DefaultPowerReduction)
+	initCoins  = sdk.NewCoins(sdk.NewCoin(umeeapp.BondDenom, initTokens))
+)
+
+// NewTestMsgCreateValidator test msg creator
+func NewTestMsgCreateValidator(address sdk.ValAddress, pubKey cryptotypes.PubKey, amt sdk.Int) *stakingtypes.MsgCreateValidator {
+	commission := stakingtypes.NewCommissionRates(sdk.ZeroDec(), sdk.ZeroDec(), sdk.ZeroDec())
+	msg, _ := stakingtypes.NewMsgCreateValidator(
+		address, pubKey, sdk.NewCoin(types.UmeeDenom, amt),
+		stakingtypes.Description{}, commission, sdk.OneInt(),
+	)
+
+	return msg
+}
+
+func (s *IntegrationTestSuite) Test_SetFeederDelegation() {
+	app, ctx := s.app, s.ctx
+
+	feederAddr := sdk.AccAddress([]byte("addr________________"))
+	feederAcc := app.AccountKeeper.NewAccountWithAddress(ctx, feederAddr)
+	app.AccountKeeper.SetAccount(ctx, feederAcc)
+
+	err := s.app.OracleKeeper.ValidateFeeder(ctx, feederAddr, valAddr)
+	s.Require().Error(err)
+
+	s.app.OracleKeeper.SetFeederDelegation(ctx, valAddr, feederAddr)
+
+	err = s.app.OracleKeeper.ValidateFeeder(ctx, feederAddr, valAddr)
+	s.Require().NoError(err)
+}
+
+func (s *IntegrationTestSuite) Test_GetFeederDelegation() {
+	app, ctx := s.app, s.ctx
+
+	feederAddr := sdk.AccAddress([]byte("addr________________"))
+	feederAcc := app.AccountKeeper.NewAccountWithAddress(ctx, feederAddr)
+	app.AccountKeeper.SetAccount(ctx, feederAcc)
+
+	s.app.OracleKeeper.SetFeederDelegation(ctx, valAddr, feederAddr)
+	s.Require().Equal(app.OracleKeeper.GetFeederDelegation(ctx, valAddr), feederAddr)
+}
+
+func (s *IntegrationTestSuite) Test_MissCounter() {
+	app, ctx := s.app, s.ctx
+	missCounter := uint64(rand.Intn(100))
+
+	s.Require().Equal(app.OracleKeeper.GetMissCounter(ctx, valAddr), uint64(0))
+	app.OracleKeeper.SetMissCounter(ctx, valAddr, missCounter)
+	s.Require().Equal(app.OracleKeeper.GetMissCounter(ctx, valAddr), missCounter)
+
+	app.OracleKeeper.DeleteMissCounter(ctx, valAddr)
+	s.Require().Equal(app.OracleKeeper.GetMissCounter(ctx, valAddr), uint64(0))
+}
+
+func (s *IntegrationTestSuite) Test_AggregateExchangeRatePrevote() {
+	app, ctx := s.app, s.ctx
+
+	prevote := types.AggregateExchangeRatePrevote{
+		Hash:        "hash",
+		Voter:       addr.String(),
+		SubmitBlock: 0,
+	}
+	app.OracleKeeper.SetAggregateExchangeRatePrevote(
+		ctx,
+		valAddr,
+		prevote,
+	)
```

Comment:
```suggestion
	app.OracleKeeper.SetAggregateExchangeRatePrevote(ctx, valAddr, prevote)
```

Reply:
Good catch. @adamewozniak we don't need to do multi-line breaks unless it exceeding the linting setting.